File: iosApp/iosApp/RootView.swift
import Shared
import SwiftUI

struct RootView: View {
    @ObservedObject
    private var childStack: ObservableValue<ChildStack<AnyObject, RootComponentChild>>

    init(component: RootComponent) {
        childStack = ObservableValue(component.childStack)
    }

    var body: some View {
        let child = self.childStack.value.active.instance

        switch child {
        case let menu as RootComponentChild.Menu:
            MenuView(
                component: menu.component
            )
        case let draw as RootComponentChild.Draw:
            DrawView(
                component: draw.component
            )
            .transition(
                .asymmetric(
                    insertion: AnyTransition.move(edge: .trailing),
                    removal: AnyTransition.move(edge: .trailing)
                )
            )
            .animation(.easeInOut)
        case let load as RootComponentChild.Load:
            LoadView(
                component: load.component
            )
            .transition(
                .asymmetric(
                    insertion: AnyTransition.move(edge: .trailing),
                    removal: AnyTransition.move(edge: .trailing)
                )
            )
            .animation(.easeInOut)
        case let save as RootComponentChild.Save:
            SaveView(
                component: save.component
            )
            .transition(
                .asymmetric(
                    insertion: AnyTransition.move(edge: .trailing),
                    removal: AnyTransition.move(edge: .trailing)
                )
            )
            .animation(.easeInOut)
        case let game as RootComponentChild.Game:
            GameView(
                component: game.component
            )
            .transition(
                .asymmetric(
                    insertion: AnyTransition.move(edge: .trailing),
                    removal: AnyTransition.move(edge: .trailing)
                )
            )
            .animation(.easeInOut)

        default: EmptyView()
        }
    }
}

File: composeApp/src/commonMain/kotlin/com/benjdero/gameoflife/RootView.kt
package com.benjdero.gameoflife

import androidx.compose.runtime.Composable
import com.arkivanov.decompose.extensions.compose.stack.Children
import com.benjdero.gameoflife.RootComponent.Child
import com.benjdero.gameoflife.draw.DrawView
import com.benjdero.gameoflife.game.GameView
import com.benjdero.gameoflife.load.LoadView
import com.benjdero.gameoflife.menu.MenuView
import com.benjdero.gameoflife.save.SaveView
import com.benjdero.gameoflife.theme.MyTheme

@Composable
fun RootView(
    component: RootComponent
) {
    MyTheme {
        Children(
            stack = component.childStack
        ) {
            when (val child: Child = it.instance) {
                is Child.Menu -> MenuView(child.component)
                is Child.Draw -> DrawView(child.component)
                is Child.Load -> LoadView(child.component)
                is Child.Save -> SaveView(child.component)
                is Child.Game -> GameView(child.component)
            }
        }
    }
}